index.html

```diff
@@ -0,0 +1,96 @@
+<!DOCTYPE html>
+<html lang="ja">
+<head>
+  <meta charset="UTF-8">
+  <meta http-equiv="X-UA-Compatible" content="IE=edge">
+  <meta name="viewport" content="width=device-width, initial-scale=1.0">
+  <title>test</title>
+  <script src="https://cdn.jsdelivr.net/npm/vue@2/dist/vue.js"></script>
+</head>
+
+<body>
+  <div id="app">
+    <h1>TODO LIST</h1>
+    <ul>
+      <li v-for="todo in todos" :key="todo.id">
+        <p>
+          {{ todo.text }}
+          <button @click="deleteTodo(todo.id)">削除</button>
+          <button @click="tryToEditTodo(todo.id)">編集</button>
+        </p>
+      </li>
+    </ul>
+    <h2>新規作成する</h2>
+    <input v-model="addText">
+    <button @click="addTodo()">TODO LISTへ追加</button><br>
+    <h2>編集する</h2>
+    <input v-model="editText">
+    <button @click="editTodo()">更新</button>
+  </div>
+
+  <script>
+    new Vue({
+      el: '#app',
+      data: function () {
+        return {
+          todos: [],
+          addText: '',
+          editId: '',
+          editText: ''
+        }
+      },
+      mounted () {
+        if (window.localStorage.getItem('todos')) {
+          this.todos = JSON.parse(window.localStorage.getItem('todos'))
+        }
+      },
+      methods: {
+        getMaxId () {
+          if (this.todos.length !== 0) {
+            const ids = this.todos.map(item => item.id)
+            return Math.max(...ids)
+          } else {
+            return 0
+          }
+        },
+        addTodo () {
+          if (this.addText !== '') {
+            this.todos.push({ text: this.addText, id: this.getMaxId() + 1 })
+            this.addText = ''
+            this.saveTodo()
+          } else {
+            window.alert('空欄でTODO LISTへ追加はできません')
+          }
+        },
+        saveTodo () {
+          const stringfied = JSON.stringify(this.todos)
+          window.localStorage.setItem('todos', stringfied)
+        },
+        deleteTodo (id) {
+          const index = this.todos.findIndex((elm) => elm.id === id)
+          this.todos.splice(index, 1)
+          this.editText = ''
+          this.editId = ''
+          this.saveTodo()
+        },
+        tryToEditTodo (id) {
+          const index = this.todos.findIndex((elm) => elm.id === id)
+          this.editText = this.todos[index].text
+          this.editId = id
+        },
+        editTodo () {
+          if (this.editText !== '') {
+            const element = this.todos.find((elm) => elm.id === this.editId)
+            element.text = this.editText
```

Comment:
具体的な名前が付けられるのであればそうしておきましょう。

```suggestion
            const todo = this.todos.find((todo) => todo.id === this.editId)
            todo.text = this.editText
```

Reply:
itemやelementからtodoに変換しておきます